kasa/iot/modules/time.py

```diff
@@ -20,20 +23,22 @@ def query(self):
     def time(self) -> datetime:
         """Return current device time."""
         res = self.data["get_time"]
-        return datetime(
+        time = datetime(
             res["year"],
             res["month"],
             res["mday"],
             res["hour"],
             res["min"],
             res["sec"],
         )
+        return time.astimezone(self.timezone)
 
     @property
-    def timezone(self):
+    def timezone(self) -> timezone | None:
```

Comment:
The smart one has a different signature and returns UTC as a fallback, should we do the same here? In any case, both should have the same signature with the base class.

Reply:
That's done